### PR TITLE
Monk 7.0

### PR DESCRIPTION
**Pull request by Skooz**
New MNK additions as discussed in https://github.com/attickdoor/XIVComboPlugin/issues/307

API was updated (at least on staging?) to include the new MNK gauge, so I went ahead and implemented MNK, if only for the sake of further discussion.

Bootshine, True Strike, and Snap Punch _**change to**_ Dragon Kick, Twin Snakes, and Demolish when no Fury charges are present. Includes lvl. 92 equivalents. 

If you wish to test it out, there's a build available over at my fork: https://github.com/Skooz/XIVComboPlugin/releases/

### CODE REVIEW

**Comment by Drimphed** [comment]
Hey just letting you know that i tested this out and it worked flawlessly. Thanks a bunch. Makes monk so much better

File: XIVComboPlugin/IconReplacer.cs [hunk 1]
```diff
@@ -764,7 +764,28 @@ private ulong GetIconDetour(byte self, uint actionID)
                 }
 
             // MONK
-            // haha you get nothing now
+            if (Configuration.ComboPresets.HasFlag(CustomComboPreset.MonkFuryCombo))
+            {
+                var gauge = JobGauges.Get<MNKGauge>();
```
Comment (attickdoor) [inline XIVComboPlugin/IconReplacer.cs:769]: I'm not sure what level of caching is happening for getting gauges, but I'd prefer to only grab this gauge for an action that it applies to. Not every single time this function runs if the combo is set.

**Comment by attickdoor** [comment]
This one seems valid to me, I commented in my issue. I'm pretty sure it fits all the guidelines.

**Comment by Skooz** [comment]
I think the only other addition I considered is combo-ing Masterful Blitz and Perfect Balance.

```
var gauge = JobGauges.Get<MNKGauge>();

if (actionID == MNK.MasterfulBlitz)
{
    if (gauge.BlitzTimeRemaining <= 0 || level < 60) return MNK.PerfectBalance;
}
```

I'm not sure if this was something discussed in the past, but the game logic supports it:
- You **need** to use Perfect Balance in order to eventually use Masterful Blitz, to begin with
- You **cannot** use Perfect Balance again until **after** you've spent your Masterful Blitz move, so there's no reason to keep it separate

I tested it out on my fork, and it feels fine... I can throw it in if you think it'd be a nice extra addition. Gets rid of 1 button.

File: XIVComboPlugin/IconReplacer.cs (hunk 1 above)
Comment (Skooz) [reply XIVComboPlugin/IconReplacer.cs]: Oops! You're right. I actually had the same thought myself, but I guess it slipped my mind by the time I was done. Switched it around in my recent commit, so it should be good now.

**Comment by attickdoor** [comment]
Yeah, the Blitz/PB thing makes sense to me. I think I had that in my mind last expac but it wasn't exactly at the top of my list of things to add.

**Comment by Skooz** [comment]
Alright! Put that in as well. Leaves us with two new options for MNK. 

image

File: XIVComboPlugin/IconReplacer.cs [hunk 2]
```diff
@@ -764,7 +764,34 @@ private ulong GetIconDetour(byte self, uint actionID)
...
+                }
+            }
+
+            if (Configuration.ComboPresets.HasFlag(CustomComboPreset.MonkPerfectBlitz))
+            {
+                if (actionID == MNK.MasterfulBlitz)
+                {
+                    if (JobGauges.Get<MNKGauge>().BlitzTimeRemaining <= 0 || level < 60) return MNK.PerfectBalance;
+                }
+            }
```
Comment (attickdoor) [inline XIVComboPlugin/IconReplacer.cs:794]:
```suggestion
            if (Configuration.ComboPresets.HasFlag(CustomComboPreset.MonkPerfectBlitz))
            {
                if (actionID == MNK.MasterfulBlitz)
                {
                    if (JobGauges.Get<MNKGauge>().BlitzTimeRemaining <= 0 || level < 60) return MNK.PerfectBalance;
                    return iconHook.Original(self, actionID);
                }
            }
```

**Comment by attickdoor** [comment]
Thinking more about it, I can see how raiders might get upset about "but what if I want to load up on Fury for damage buffs", and I think my answer to that is "get bent".

**Comment by Skooz** [comment]
> Thinking more about it, I can see how raiders might get upset about "but what if I want to load up on Fury for damage buffs", and I think my answer to that is "get bent".

I imagine the sweaty folk will find some way to optimize Fury charges with Six-sided Star usage, ensuring they go into buff phases with a particular set of charges without purposefully overcapping and dropping potency. 

Telling people to purposefully overcap on Coeurl Fury charges is in that "cap your fps" territory of gameplay. If you're going to those lengths, then you're probably willing to just turn off XIVCombo if you don't like it.

**Comment by attickdoor** [comment]
Alright, let's do it.

**Pull request merged**